scripts/ban-unordered-abi.sh

```
@@ -0,0 +1,48 @@
+files=$(rg "${RG_ARGS[@]}" -l -g'*.rs' "$(printf '%s|' "${ABI_HINTS[@]}")" crates/ || true)
+
+if [[ -z "${files}" ]]; then
+  echo "ban-unordered-abi: no ABI-ish files found (by heuristic). OK."
+  exit 0
+fi
+
+echo "ban-unordered-abi: scanning ABI-ish Rust files..."
+violations=0
+
+# HashMap/HashSet are not allowed in ABI-ish types. Use Vec<(K,V)> sorted, BTreeMap, IndexMap with explicit canonicalization, etc.
+if rg "${RG_ARGS[@]}" -n -S '\b(HashMap|HashSet)\b' $files; then
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)